### PR TITLE
accumulate into vector and assert, instead of printing

### PR DESCRIPTION
**Pull request by matthew-piziak**
I'm only making this change in one place so that people can express
their preferences for this stylistic change. If/when this change is
approved I'll go ahead and translate the rest of the `std::ops`
examples.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Ah nice, thanks!

@bors: r+ rollup

File: src/libcore/iter/range.rs [hunk 1]
```diff
@@ -295,20 +295,8 @@ impl<A: Step> ops::Range<A> {
...
-    ///
-    /// This prints:
-    ///
-    /// ```text
-    /// 0
-    /// 2
-    /// 4
-    /// 6
-    /// 8
+    /// let result: Vec<_> = (0..10).step_by(2).take(10).collect();
```
Comment (Limeth) [inline src/libcore/iter/range.rs]: Shouldn't this have `.take(5)` instead?

**Comment by matthew-piziak** [comment]
@Limeth Actually this example doesn't need a `take` call at all. Fixed. :)

r? @GuillaumeGomez 

@steveklabnik

**Comment by GuillaumeGomez** [comment]
I learnt something new today. Thanks again!

@bors: r+ rollup

**Pull request merged**